packages/kbn-es/src/utils/wait_until_cluster_ready.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Client } from '@elastic/elasticsearch';
+import { ToolingLog } from '@kbn/tooling-log';
+const DEFAULT_READY_TIMEOUT = 60 * 1000; // 1 minute
+
+export interface WaitOptions {
+  client: Client;
+  log: ToolingLog;
+  readyTimeout?: number;
+}
+
+/**
+ * General method to wait for the ES cluster status to be green
+ */
+export async function waitUntilClusterReady({
+  client,
+  log,
+  readyTimeout = DEFAULT_READY_TIMEOUT,
+}: WaitOptions) {
+  let attempt = 0;
+  const start = Date.now();
+
+  log.info('waiting for ES cluster to report a green status');
+
+  while (true) {
+    attempt += 1;
+
+    try {
+      const resp = await client.cluster.health();
+      if (resp.status === 'green') {
+        return;
+      }
```

Comment:
Original methods waits for status to be yellow or green, but I believe to avoid flakiness we better  not tolerate `yellow` and wait for the `green` one. Thoughts?

Reply:
Thank you, let's start with green and see if we need to adjust later.